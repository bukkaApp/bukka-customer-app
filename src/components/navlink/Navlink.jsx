import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './navlink.scss';

const Navlink = ({ text, href, classNames, children, ...props }) => (
  <Link to={href} className={classNames} {...props}>
    {text || children}
  </Link>
);

export default Navlink;

Navlink.defaultProps = {
  children: '',
  text: ''
};

Navlink.propTypes = {
  text: PropTypes.string,
  href: PropTypes.string.isRequired,
  classNames: PropTypes.string.isRequired,
  children: PropTypes.node,
};
