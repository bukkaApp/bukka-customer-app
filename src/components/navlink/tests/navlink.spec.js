import React from 'react';
import Navlink from '../Navlink';

describe('Navlink component', () => {
  const props = { text: 'signIn', href: '/back', classNames: '' };
  const wrapper = shallow(<Navlink {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
