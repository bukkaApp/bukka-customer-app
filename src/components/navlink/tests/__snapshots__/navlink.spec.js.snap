// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navlink component renders properly 1`] = `ShallowWrapper {}`;
